Fix index_reduce mean error when dtype is bfloat16 (#139242)

Kattos opened the pull request:
Change the variable `count` from scalar_t to int64_t in case of accumulation error

Fixes #139242

Kattos committed:
Fix index_reduce mean error when dtype is bfloat16 (#139242)

Change the variable `count` from scalar_t to int64_t in case of accumulation error

Kattos commented:
@pytorchbot label "topic: not user facing"

eqy reviewed (commented):
Could you add the failing case as a test please?

Kattos committed:
Fix index_reduce mean error when dtype is bfloat16 (#139242)

Change the variable `count` from scalar_t to int64_t in case of accumulation error

Kattos committed:
Fix index_reduce mean error when dtype is bfloat16 (#139242)

Change the variable `count` from scalar_t to int64_t in case of accumulation error

Kattos commented:
No problem. I have added some tests but I'm not sure whether the modifications were made in the right places.

eqy commented:
> No problem. I have added some tests but I'm not sure whether the modifications were made in the right places.

maybe move to `test/test_indexing.py`?
